### ISSUE TITLE
libressl 3.7.1 (libcrypto.so.50.2.0) breaks perl Net::DNS::SEC DSA verify

### ISSUE THREAD

**Issue by rwfranks** [libressl/portable#861]
Net::DNS::SEC 1.20
Net::DNS 1.37
perl 5.36.1

Pre-install test t/31-DSA-SHA1.t works with libressl 3.7.0

Using libressl-3.7.1, libcrypto EVP_DigestVerify returns -1 internal error: 
```
ok 8 - signature created using private key
not ok 9 - signature verified using public key
#   Failed test 'signature verified using public key'
#   at test.pl line 118.
#          got: '-1'
#     expected: '1'
```
Fault is also present in libressl-3.7.2

Error stack at point of failure contains:
```
140092269976640:error:0AFFF070:dsa routines:CRYPTO_internal:invalid parameters:/tmp/libressl-3.7.2/crypto/dsa/dsa_lib.c:492:
140092269976640:error:0AFFF003:dsa routines:CRYPTO_internal:BN lib:/tmp/libressl-3.7.2/crypto/dsa/dsa_ossl.c:417:
```

**Comment by botovq** [comment]
The reason for this is that we no longer accept a DSA with a private key
of 0 since this may lead to DoS.  See here for one of the issues that
led to this change:

https://github.com/openssl/openssl/issues/20268

The workaround is easy: do not set a missing private key to zero, but
rather omit it. That's what this patch does. It should fix the test.

https://github.com/openbsd/ports/blob/master/net/p5-Net-DNS-SEC/patches/patch-SEC_xs

**Comment by rwfranks** [comment]
On Sun, 21 May 2023 at 14:15, Theo Buehler ***@***.***> wrote:
>
> The reason for this is that we no longer accept a DSA with a private key
> of 0 since this may lead to DoS. See here for one of the issues that
> led to this change:
>
> https://github.com/openssl/openssl/issues/20268
>
> The workaround is easy: do not set a missing private key to zero, but
> rather omit it. That's what this patch does. It should fix the test.
>
> https://github.com/openbsd/ports/blob/master/net/p5-Net-DNS-SEC/patches/patch-SEC_xs
>

Indeed it does.
Fix will be in Net::DNS::SEC 1.21

Thanks for the prompt response

--RWF
